common/src/main/java/org/dash/wallet/common/util/GenericUtils.java

```diff
@@ -50,4 +54,92 @@ public static void showToast(Context context, String messages) {
+        Locale defaultLocale = new Locale(deviceLocaleLanguage, countryCode);
+
+        Currency defaultCurrency = Currency.getInstance(defaultLocale);
+        return defaultCurrency.getCurrencyCode();
+    }
+
+    /**
+     * Funtion which returns a concatenation of the currency code together with the currency symbol
+     * For currencies used by multiple countries, we set a locale with any country using the currency
+     * If the currentCurrencySymbol equals the currency code, we just use the currency code, otherwise we
+     * concatenate both
+     * @param currencyCode
+     * @return
+     */
+    public static String setCurrentCurrencySymbolWithCode(String currencyCode) {
+        Locale currentLocale = new Locale("", "");
+        String currentCurrencySymbol = "";
+        switch (currencyCode.toLowerCase(Locale.ROOT)) {
+            case "eur":
+                currentLocale = Locale.FRANCE;
+                break;
+            case "xof":
+                currentLocale = new Locale("fr", "CM");
+                break;
+            case "xaf":
+                currentLocale = new Locale("fr", "SN");
+                break;
+            case "cfp":
+                currentLocale = new Locale("fr", "NC");
+                break;
+            case "hkd":
+                currentLocale = new Locale("en", "HK");
+                break;
+            case "bnd":
+                currentLocale = new Locale("ms", "BN");
+                break;
+            case "aud":
+                currentLocale = new Locale("en", "AU");
+                break;
+            case "gbp":
+                currentLocale = Locale.UK;
+                break;
+            case "inr":
+                currentLocale = new Locale("en", "IN");
+                break;
+            case "nzd":
+                currentLocale = new Locale("en", "NZ");
+                break;
+            case "ils":
+                currentLocale = new Locale("iw", "IL");
+                break;
+            case "jod":
+                currentLocale = new Locale("ar", "JO");
+                break;
+            case "rub":
+                currentLocale = new Locale("ru", "RU");
+                break;
+            case "zar":
+                currentLocale = new Locale("en", "ZA");
+                break;
+            case "chf":
+                currentLocale = new Locale("fr", "CH");
+                break;
+            case "try":
+                currentLocale = new Locale("tr", "TR");
+                break;
+            case "usd":
+                currentLocale = Locale.US;
+                break;
+        }
+        currentCurrencySymbol = TextUtils.isEmpty(currentLocale.getLanguage()) ?
+                currencySymbol(currencyCode.toLowerCase(Locale.ROOT)) : Currency.getInstance(currentLocale).getSymbol();
+
+        return currencyCode.equalsIgnoreCase(currentCurrencySymbol) ? currencyCode :
+                currencyCode.concat(" ").concat(currentCurrencySymbol);
```

Comment:
I remember getting a warning about concatenating strings that were like this `a + " " + b`, where `a` and `b` were Strings.

The warning said there would be problems rendering the string in come locales that might use RTL.  Not quite sure.  so I created a string "%s %s" in the resources.  I don't know if that is necessary in this case.

Reply:
I have question about the designs regarding the need for this method.

Reply:
We will leave this code in for now and address it in a different story concerned with formats of values.